reframe/frontend/statistics.py

```diff
@@ -66,46 +67,113 @@ def retry_report(self):
 
         return '\n'.join(report)
 
+    def json(self, reframe_info=None, force=False):
```

Comment:
This function should only be responsible for constructing the json report for each run. It should not be involved with the `reframe_info` at all.